### PR TITLE
libtiff: update to 4.5.1

### PR DESCRIPTION
**Pull request by bgilbert**
Upstream added a Windows resource file and refactored endian detection.

### CODE REVIEW

File: subprojects/packagefiles/libtiff/libtiff/meson.build [hunk 1]
```diff
@@ -57,6 +57,13 @@ sources = files(
   'tif_@0@.c'.format(host_machine.system() == 'windows' ? 'win32' : 'unix'),
 )
 
+if host_machine.system() == 'windows' and meson.get_compiler('c').get_argument_syntax() == 'msvc'
```
Comment (neheb) [inline subprojects/packagefiles/libtiff/libtiff/meson.build:60]: Not compatible with MSYS2?

Comment (bgilbert) [reply subprojects/packagefiles/libtiff/libtiff/meson.build:60]: Personally I'd make the check less restrictive, but to be safe, I'm mirroring the one in the upstream CMakeLists.

**Comment by bgilbert** [comment]
Okay, this is ready for review.

File: subprojects/packagefiles/libtiff/libtiff/meson.build (hunk 1 above)
Comment (neheb) [reply subprojects/packagefiles/libtiff/libtiff/meson.build:60]: Fishy. I'd try just testing for windows.

Comment (bgilbert) [reply subprojects/packagefiles/libtiff/libtiff/meson.build]: Okay, done.

**Pull request merged**